### PR TITLE
897743 - MAUI - Prepare UG documentation for edit element appearance customization and incremental loading support

### PR DESCRIPTION
**Pull request by AbhishekSF4201**
- have included the Value member path, drop down suggestion, can filter suggestions and incremental loading

### CODE REVIEW

**Review by ashok-kuvaraja — CHANGES_REQUESTED** [review]

File: MAUI/DataGrid/Column-types.md [hunk 1]
```diff
@@ -877,6 +877,169 @@ dataGrid.Columns.Add(comboBoxColumn);
 
 N> The `ShowClearButton` property has no effect in non-editable mode..
 
+### Drop down width
```
[the remark this answers is not in the document]
Comment (AbhishekSF4201) [reply MAUI/DataGrid/Column-types.md:880]: changed the title.

File: MAUI/DataGrid/Column-types.md [hunk 2]
```diff
@@ -877,6 +877,169 @@ dataGrid.Columns.Add(comboBoxColumn);
...
+
+The ComboBox control include a `DropDownWidth` that allows the user to increase the width of the drop down box.
```
[the remark this answers is not in the document]
Comment (AbhishekSF4201) [reply MAUI/DataGrid/Column-types.md:882]: changed the content

File: MAUI/DataGrid/Column-types.md [hunk 3]
```diff
@@ -877,6 +877,169 @@ dataGrid.Columns.Add(comboBoxColumn);
...
+     this.CustomerID = Customers.ToObservableCollection();
+}
+{% endhighlight %}
+{% endtabs %}
+
+<img alt="DropDownWidth" src="Images\column-types\maui-datagrid-comboBox-drop-down-width.png" width="404"/>
+
+### Can filter suggestions
+
+The `CanFilterSuggestion` property in the SfComboBox component is used to enable or disable the filtering of suggestions based on user input. When set to `true`, this property allows the combo box to filter the list of suggestions dynamically as the user types into the input field.
```
[the remark this answers is not in the document]
Comment (AbhishekSF4201) [reply MAUI/DataGrid/Column-types.md:906]: done

File: MAUI/DataGrid/performance.md [hunk 4]
```diff
@@ -36,4 +36,49 @@ N> Download demo application from [GitHub](https://github.com/SyncfusionExamples
...
+
+# Incremental loading
+
+DataGrid supports to load the data incrementally using `ISupportIncrementalLoading` interface. `ISupportIncrementalLoading` interface has `LoadMoreItemsAsync` method which helps to load the data incrementally. `LoadMoreItemsAsync` called in on-demand while scrolling based on `HasMoreItems` property.
+
+If `HasMoreItems` is false, SfDataGrid stops calling `LoadMoreItemsAsync`. SfDataGrid have `IncrementalList` which is derived from `ISupportIncrementalLoading`. You can use IncrementalList or create collection derived from `ISupportIncrementalLoading` and bind it `SfDataGrid.ItemsSource`.
+
+Additionally it supports `SfDataGrid.DataFetchSize` , which will automatically fetch the data when the user reaches the end of the  `dataGrid`. 
+
+In the below code, `IncrementalList` is initialized by passing Action to its constructor for loading items incrementally.
```
Comment (ashok-kuvaraja) [inline MAUI/DataGrid/performance.md:50]: The DataGrid supports loading data incrementally using the `ISupportIncrementalLoading` interface. This interface includes the `LoadMoreItemsAsync` method, which helps to load data incrementally. The `LoadMoreItemsAsync` method is called on-demand while scrolling, based on the `HasMoreItems` property.

If `HasMoreItems` is false, SfDataGrid stops calling `LoadMoreItemsAsync`. SfDataGrid includes `IncrementalList`, which is derived from `ISupportIncrementalLoading`. You can use `IncrementalList` or create a collection derived from `ISupportIncrementalLoading` and bind it to `SfDataGrid.ItemsSource`.

Additionally, the `SfDataGrid.DataFetchSize` property defines the count argument in the `LoadMoreItemsAsync` method, which determines the amount of data to be loaded into the items source when the user reaches the end of the `SfDataGrid`

In the code below, `IncrementalList` is initialized by passing an Action to its constructor for loading items incrementally.

**Review by ashok-kuvaraja — CHANGES_REQUESTED** [review]

**Review by ashok-kuvaraja — APPROVED** [review]
The changes are fine.

**Pull request merged**